modules/nlu/src/backend/prediction-handler.ts

```diff
@@ -36,14 +38,24 @@ export class PredictionHandler {
       .mapValues(({ model }) => model)
       .value()
 
-    const detectedLanguage = await this.engine.detectLanguage(textInput, loadedModels)
+    let detectedLanguage: string | undefined
+    try {
+      detectedLanguage = await this.engine.detectLanguage(textInput, loadedModels)
+    } catch (err) {
+      let msg = `An error occured when detecting language for input "${textInput}"\n`
+      msg += `Falling back on default language: ${this.defaultLanguage}.`
+      this.logger.attachError(err).error(msg)
+    }
 
-    let nluResults: sdk.IO.EventUnderstanding | undefined
+    let nluResults: WithoutDetectedLanguage | undefined
 
-    const languagesToTry = _.uniq([detectedLanguage, this.anticipatedLanguage, this.defaultLanguage])
+    const languagesToTry = _([detectedLanguage, this.anticipatedLanguage, this.defaultLanguage])
+      .filter(l => !_.isUndefined(l))
+      .uniq()
+      .value()
 
     for (const lang of languagesToTry) {
-      nluResults = await this.tryPredictInLanguage(textInput, includedContexts, lang)
+      nluResults = await this.tryPredictInLanguage(textInput, lang)
```

Comment:
hmm I feel this is a bit weird. maybe we overlooked that in previous PRs

if languagesToTry actually contains more than a single language and engine has all language predictors in memory for instance `[fr, es]` and that the actual language is really `fr`. From what I understand, we're predicting for fr then for `es`, overwrite `nluResults` and return the `es` results in the end ? Even if `fr` prediction was the best guess. 

Am I crazy ?

Reply:
Yes you are 😂

Language `es` won't override predictions for `fr` because we only fallback on `es` if `fr` failed.

You'll see in the loop:

```
if (!this.isEmptyOrError(nluResults)) {
    break
}
```

Reply:
this part of the code is actually quite well tested by the `prediction-handler.test.ts` test file which is super hard to maintain (because of its high-levelness), but also quite usefull as it helped me caught few bugs in the past.

Reply:
wow yes I am crazy I didn't see the ! . Thanks for exposing my adhd 🤣